### PR TITLE
[PROTOCOL-657] Dynamically deploy contracts with network specific dapp addresses

### PR DESCRIPTION
**Pull request by sypeer**
The following facets have had constructor arguments added to save the network specific dapp addresses as an `immutable` variable:
- Sushiswap Facet 
- Uniswap Facet
- Aave Facet

LibUniswap and LibSushiswap have been merged into LibSwapper

TTokenAaveStrategy has an added init arg to store the LP address provider address and uses the updated LibDapps to grab the relevant Aave lending pool

### CODE REVIEW

File: config/index.ts [hunk 1]
```diff
@@ -61,6 +62,8 @@ export const getNativeToken = (network: Network) => {
   return wrappedNativeToken
 }
 
+export const getDappAddresses = (network: Network) => dapps[network.name]
```
Comment (passabilities) [inline config/index.ts:65]: should call `getNetworkName(network)`

File: deploy/markets.ts [hunk 2]
```diff
@@ -129,6 +130,7 @@ const initializeMarkets: DeployFunction = async (hre) => {
                 case 'TokenSymbol':
                   return tokenAddresses.all[value]
                 case 'Address':
+                  return dappAddresses[value]
```
Comment (passabilities) [inline deploy/markets.ts:133]: this is good for now but will probably have to rethink things when we want to use addresses for other things

**Review by passabilities — CHANGES_REQUESTED** [review]

File: config/index.ts (hunk 1 above)
Comment (sypeer) [reply config/index.ts]: fixed

**Review by passabilities — APPROVED** [review]

File: deploy/markets.ts (hunk 2 above)
Comment (sypeer) [reply deploy/markets.ts:133]: can add additional types? `AddressProviderConstant` as an additional type

Comment (sypeer) [reply deploy/markets.ts:133]: Added an additional type `ProtocolAddressConstant` for now

**Review by dumebi — APPROVED** [review]

**Pull request merged**